### PR TITLE
HDDS-8408. [Snapshot] Implement 'ozone fs' command for deleting snapshots

### PR DESCRIPTION
**Pull request by aswinshakil**
## What changes were proposed in this pull request?

To implement delete snapshot file system commands.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8408

## How was this patch tested?

New unit test.

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java [hunk 1]
```diff
@@ -285,6 +285,19 @@ public void testDeleteBucketWithSnapshot() throws Exception {
     Assertions.assertTrue(listSnapKeyOut.contains(snapshotKeyPath));
   }
 
+  @Test
+  public void testSnapshotDeleteSuccessAndFailure() throws Exception {
```
Comment (hemantk-12) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java:289]: nit: I prefer to test only one thing in a test. Also by making `deleteSnapshot()` another function you are delegating test to another function. Better way would have been to have two different test one for success and one for failure.

**Review by hemantk-12 — COMMENTED** [review]
LGTM.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 2]
```diff
@@ -938,6 +938,12 @@ public Path createSnapshot(Path path, String snapshotName)
         OM_SNAPSHOT_INDICATOR + OZONE_URI_DELIMITER + snapshot);
   }
 
+  @Override
+  public void deleteSnapshot(Path path, String snapshotName)
+      throws IOException {
+    adapter.createSnapshot(pathToKey(path), snapshotName);
```
Comment (smengcl) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:944]: copy pasta?
```suggestion
    adapter.deleteSnapshot(pathToKey(path), snapshotName);
```

Comment (aswinshakil) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:944]: Ah yes my bad. Will update.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java (hunk 1 above)
Comment (aswinshakil) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java:289]: Understood. The test was pretty small so added it to the same test.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java [hunk 3]
```diff
@@ -346,4 +362,22 @@ private String createSnapshot() throws Exception {
 
     return snapshotName;
   }
+
+  private void deleteSnapshot(String snapshotName) throws Exception {
```
Comment (hemantk-12) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java:366]: It doesn't have to be a separate function. Please move it to the test itself.

Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java]: Thanks @aswinshakil  for fixing it.

**Review by hemantk-12 — APPROVED** [review]

**Review by prashantpogde — APPROVED** [review]
Looks good to me! thank you for making these changes @aswinshakil

**Pull request merged**